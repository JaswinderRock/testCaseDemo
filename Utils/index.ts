export const findByTestAttr = (component: any, attr: String) => {
    const wrapper = component.find(`[data-test='${attr}']`);
    return wrapper;
}
export const findByClassName = (component: any, className: String) => {
    const wrapper = component.find(`.${className}`);
    return wrapper;
}