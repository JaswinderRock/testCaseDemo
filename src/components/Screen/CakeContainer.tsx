import React, { Component } from 'react';
import { connect } from 'react-redux';
import { buyCake } from '../Action/action';

type CakeProps = {
    numOfCakes: number;
    buyCake: () => void;
};
class CakeContainer extends Component<CakeProps>{
    render() {
        return (
            <div style={{ display: 'flex' }}>
                <div style={{ margin: 'auto' }}>
                    <h2>Number of Cakes {this.props.numOfCakes}</h2>
                    <button style={{ display: 'flex', margin: 'auto', borderRadius: '4px', textAlign: 'center', color: "white", background: 'blue', padding: '10px', fontSize: '24px', }} onClick={this.props.buyCake}>Buy Cake</button>
                </div>
            </div>
        )

    }
}
function mapStateToProps(state: any) {
    return { numOfCakes: state.numberOfCakes };
}
export default connect(mapStateToProps, { buyCake })(CakeContainer);