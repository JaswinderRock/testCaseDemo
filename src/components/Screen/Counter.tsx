import React from "react";
import { Component } from "react";
type CounterProps = {
    message: string
}
type CounterState = {
    count: number
}
export class Counter extends Component<CounterProps, CounterState> {
    state = {
        count: 0,
    }
    handleClick = () => {
        this.setState((prevState) => ({ count: prevState.count + 1 }))
    }
    render() {
        return (
            <div className="counterwrapper" style={{ display: 'flex' }}>
                <div style={{ margin: 'auto' }}>
                    <div data-test='counterdescription' className="counterdescription" style={{ marginTop: '30px' }}>
                        <h2 data-test="label-text">{this.props.message} {this.state.count}</h2> 
                    </div>
                    <button data-test="counter-btn" className="counterBtn" style={{ display: 'flex', margin: 'auto', borderRadius: '4px', marginTop: '10px', textAlign: 'center', color: "white", background: 'blue', padding: '10px', fontSize: '24px', }} onClick={this.handleClick}>Increment</button>
                </div>
            </div>
        )
    }
}