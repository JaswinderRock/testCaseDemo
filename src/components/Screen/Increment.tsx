import React, { Component } from "react";
import { increment, decrement } from "../Action/action";
import '../css/increment.css';
import { connect } from "react-redux";
import { State } from "../Reducer/reducer";

type IncrementState = {
    count: number;
    increment: () => void;
    decrement: () => void;
};
type incrementProps = {
    handleClick: () => void
}
class IncrementDecrement extends Component<IncrementState, incrementProps> {
    handleClick = () => {
        this.props.increment()
    }
    handleDecrement = () => {
        this.props.decrement()
    }
    render(): React.ReactNode {
    return (
        <div className="IncrementDecrement">
            <h1 data-test="Count-Text" className="Count-Text">Count: {this.props.count}</h1>
            <button data-test="increment-btn" className="Increment-btn" type="button" style={{ background: 'blue', color: 'white', fontSize: '18px', borderRadius: '4px', padding: '8px' }} onClick={this.handleClick}>Increment</button>
            <button data-test="decrement-btn" className="decrement-btn" style={{ background: 'blue', color: 'white', fontSize: '18px', marginLeft: '10px', borderRadius: '4px', padding: '8px' }} onClick={this.handleDecrement}>Decrement</button>
        </div>
    );
    }
};

function mapStateToProps(state: State) {
    return { count: state.count };
}
export default connect(mapStateToProps, { increment, decrement })(IncrementDecrement);
