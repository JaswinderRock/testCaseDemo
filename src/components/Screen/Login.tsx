import React, { Component } from 'react';
export class Login extends Component {
    render() {
        return (
            <div>
                <label>Login Here!</label>
            </div>
        )
    }
}