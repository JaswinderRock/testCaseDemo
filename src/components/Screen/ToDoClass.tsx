import React, { Component } from "react";
import { connect } from "react-redux";
import { setName } from '../Action/action';

type TodoProps = {
    setName: (a: string) => void;
    reduxList: Array<String>;
};
type TodoState = {
    name: String;
};
class ToDoClass extends Component<TodoProps, TodoState> {
    state = {
        name: "",
    };
    handleClick = () => {
        if (!this.state.name.trim()) {
            alert("Empty text can't be added into redux");
        }
        this.props.setName(this.state.name);
        this.setState({ name: "" });
    };
    render() {
        return (
            <div style={{ display: 'flex' }}>
                <div style={{ margin: 'auto' }}>
                    <h3 style={{ marginTop: '60px' }}>Hi i am todo</h3>
                    <input
                        style={{ padding: '10px' }}
                        value={this.state.name}
                        type="text"
                        placeholder="enter your list value here! "
                        onChange={(e) => this.setState({ name: e.target.value })}
                    />
                    <br />
                    <button
                        type="button"
                        onClick={this.handleClick}
                        style={{ height: 45, width: 120, backgroundColor: "blue", color: 'white', marginTop: '10px', fontSize: '20px', borderRadius: '4px' }}
                    >
                        Submit
                    </button>
                    <br />
                    {this.props.reduxList.map((item, index) => (
                        <div key={index} style={{ fontSize: '20px' }}>
                            {item} <br />
                        </div>
                    ))}
                </div>
            </div>
        );
    }
}
function mapStateToProps(state: any) {
    return { reduxList: state.nameList };
}
export default connect(mapStateToProps, { setName })(ToDoClass);
