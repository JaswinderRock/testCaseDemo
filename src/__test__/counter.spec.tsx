import React, { Component } from 'react';
import { shallow } from "enzyme"
import { Counter } from '../components/Screen/Counter';
import { findByClassName, findByTestAttr } from '../../Utils/index';

const setUp = () => {
    const component = shallow(<Counter message={`The Count Value is `} />);
    return component;
}

describe("Render Counter", () => {
    let component: any;
    beforeEach(() => {
        component = setUp();
    }),
    test("Counter test is render ", () => {
        expect(component).toMatchSnapshot()
    }),
        test("Should check for ClassName ", () => {
            const wrapper = findByClassName(component, 'counterwrapper')
            expect(wrapper.length).toBe(1)
        }),
        test("Should check for the header tag ", () => {
            const wrapper = findByTestAttr(component, 'counterdescription')
            expect(wrapper.length).toBe(1)
        }),
        test("Should check for button ClassName ", () => {
            const wrapper = findByClassName(component, 'counterBtn')
            expect(wrapper.length).toBe(1)
        }),
        test("renders correctly with initial count", () => {
            const labelText = findByTestAttr(component, 'label-text');
            const expectedText = "The Count Value is  0";
            const receivedText = labelText.find("h2").text().trim();
            expect(receivedText).toContain(expectedText);
        }),
        it('click the counter button to increment the value ', () => {
            const counterButton = findByTestAttr(component, 'counter-btn');
            expect(counterButton.exists()).toBe(true);
            counterButton.simulate('click');
            const expectedText = "The Count Value is  1";
            const labelText = findByTestAttr(component, 'label-text');
            const receivedText = labelText.find("h2").text().trim();
            expect(receivedText).toContain(expectedText);
        })
})