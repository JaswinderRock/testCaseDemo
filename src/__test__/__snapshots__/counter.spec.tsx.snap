// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Counter Counter test is render  1`] = `
<div
  className="counterwrapper"
  style={
    Object {
      "display": "flex",
    }
  }
>
  <div
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <div
      className="counterdescription"
      data-test="counterdescription"
      style={
        Object {
          "marginTop": "30px",
        }
      }
    >
      <h2
        data-test="label-text"
      >
        The Count Value is 
         
        0
      </h2>
    </div>
    <button
      className="counterBtn"
      data-test="counter-btn"
      onClick={[Function]}
      style={
        Object {
          "background": "blue",
          "borderRadius": "4px",
          "color": "white",
          "display": "flex",
          "fontSize": "24px",
          "margin": "auto",
          "marginTop": "10px",
          "padding": "10px",
          "textAlign": "center",
        }
      }
    >
      Increment
    </button>
  </div>
</div>
`;
